src/thunderbots/software/ai/hl/stp/play/kickoff_friendly_play.cpp

```diff
@@ -0,0 +1,151 @@
+
+    // the following positions are in the same order as the positions shown above,
+    // excluding the goalie for part 1 of this play
+    std::vector<Point> kickoff_setup_positions = {
+        // Robot 1
+        Point(world.field().centerPoint() +
+              Point(-world.field().centreCircleRadius(), 0)),
+        // Robot 2
+        // Goalie positions will be handled by the goalie tactic
+        // Robot 3
+        Point(world.field().centerPoint() +
+              Point(-world.field().centreCircleRadius() - 4 * ROBOT_MAX_RADIUS_METERS,
+                    -1.0 / 3.0 * world.field().width())),
+        // Robot 4
+        Point(world.field().centerPoint() +
+              Point(-world.field().centreCircleRadius() - 4 * ROBOT_MAX_RADIUS_METERS,
+                    1.0 / 3.0 * world.field().width())),
+        // Robot 5
+        Point(world.field().friendlyGoalpostPos().x() +
+                  world.field().defenseAreaLength() + 2 * ROBOT_MAX_RADIUS_METERS,
+              world.field().friendlyGoalpostPos().y()),
+        // Robot 6
+        Point(world.field().friendlyGoalpostNeg().x() +
+                  world.field().defenseAreaLength() + 2 * ROBOT_MAX_RADIUS_METERS,
+              world.field().friendlyGoalpostNeg().y()),
+    };
+
+    // move tactics to use to move to positions defined above
+    std::vector<std::shared_ptr<MoveTactic>> move_tactics = {
+        std::make_shared<MoveTactic>(true), std::make_shared<MoveTactic>(true),
+        std::make_shared<MoveTactic>(true), std::make_shared<MoveTactic>(true),
+        std::make_shared<MoveTactic>(true)};
+
+    // specific tactics
+    auto goalie_tactic = std::make_shared<GoalieTactic>(
+        world.ball(), world.field(), world.friendlyTeam(), world.enemyTeam());
+    auto chip_tactic = std::make_shared<ChipTactic>(world.ball(), true);
+
+    // Part 1: setup state (move to key positions)
+    while (world.gameState().isSetupState())
+    {
+        auto enemy_threats = Evaluation::getAllEnemyThreats(
+            world.field(), world.friendlyTeam(), world.enemyTeam(), world.ball(), false);
+
+        goalie_tactic->updateParams(
+            world.ball(), world.field(), world.friendlyTeam(), world.enemyTeam(),
+            enemy_threats.empty() ? std::nullopt
+                                  : std::make_optional(enemy_threats.at(0)));
+        std::vector<std::shared_ptr<Tactic>> result = {goalie_tactic};
+
+        // setup 5 kickoff positions in order of priority
+        for (int i = 0; i < kickoff_setup_positions.size(); i++)
+        {
+            move_tactics.at(i)->updateParams(kickoff_setup_positions.at(i), Angle::half(),
+                                             0);
+            result.emplace_back(move_tactics.at(i));
+        }
+
+        // yield the Tactics this Play wants to run, in order of priority
+        yield(result);
+    }
+
+    // Part 2: ready state (chip the ball)
+    while (world.gameState().isReadyState())
```

Comment:
I think a better condition here would be `!world.gameState().isPlaying()` because the gamestate switches to `Play` once the ball is kicked (99% sure), so that way we would finish the play once the ball is kicked.

Reply:
fixed, iirc those states come after each other, after `isReadyState()` is false, `isPlaying()` becomes true so either way the play would be finished.